### ISSUE TITLE
[REMOVE] Remove from scripts the deprecated methods of JodaCompatibleZonedDateTime

### ISSUE THREAD

**Issue by reta** [opensearch-project/OpenSearch-Dashboards#1600]
**Is your feature request related to a problem? Please describe.**

There are number of deprecated methods in `JodaCompatibleZonedDateTime` which are still available in scripts. Those should be removed since in some cases they pose significant performance overhead (due to deprecation warnings).

**Describe the solution you'd like**
Remove from scripts the deprecated methods from JodaCompatibleZonedDateTime. 

**Describe alternatives you've considered**

 Leave the methods there.

**Additional context**

See please https://github.com/opensearch-project/OpenSearch/issues/3156 and https://github.com/opensearch-project/OpenSearch/issues/3345

**Comment by reta** [comment]
@kavilla fyi

**Comment by kavilla** [comment]
Hello @reta ,

Thanks for opening. I have made it a bug.

Is there a documented list of `instead of x please use y` publicly available? This will help the community migrate to the proper scripts.

**Comment by reta** [comment]
Hello @kavilla ,

> Is there a documented list of instead of x please use y publicly available? This will help the community migrate to the proper scripts.

Absolutely, first of all, there are deprecation warnings in logs:
``` 
Use of the joda time method [getMillis()] is deprecated.  Use [toInstant().toEpochMilli()] instead.
```

Interestingly, there is nothing in the Elasticsearch documentation about that, just the pull request [1].
Also, there is an original issue on Elasticsearch [2]. 

[1] https://github.com/elastic/elasticsearch/pull/33486
[2] https://github.com/elastic/elasticsearch/issues/35754

**Comment by kavilla** [comment]
> Absolutely, first of all, there are deprecation warnings in logs:

Yeah, unfortunately for end users that purely use OpenSearch Dashboards to build these scripts wouldn't be able to see these messages unless they look at their logs as you stated which might not be likely. And do we know if these warning messages are enabled by default in a release (production) environment? And if it has gotten removed then when 3.0 comes then might wonder why their painless scripts no longer work.

I think we we can flag the original PR with "needs-documentation" it will create an issue in the documentation website repo to add these details. I think we need painless scripts logs in general and we can include the `this was removed in 3.0` sections or `will be removed in 3.0`

cc: @ahopp, @saratvemulapalli

**Comment by kavilla** [comment]
With https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1607, I have updated the stuff we utilized for sample data and tests so I have removed the bug label and re-added the enhancement.

Right now, we don't have a lot of control of the actual values within scripted fields and I wouldn't recommend changing this value. However, based on the above mention PR'd this is an example of the kind of impact end-users will have to do if they utilized scripted fields (which is a huge feature in the application) and they are using deprecated methods that were removed.

Since this is a breaking change on the OpenSearch side, and the deprecation message has been obfuscated by OpenSearch Dashboards not outputting it to the screen. Users would have had to check their logs to see they should have been using a different method, which isn't likely if everything is working properly.

I would prefer we give in application notification about the removal of deprecated methods in Painless Scripts and provide a link to some view that's like `if using x then use y` or something like that which is why I tagged @opensearch-project/opensearch-ux to see if they could provide some help with this.
